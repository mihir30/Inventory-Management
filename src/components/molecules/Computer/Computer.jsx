import React, { useState } from 'react';
import Processor from '../Processor/Processor';
import AddOutlinedIcon from '@mui/icons-material/AddOutlined';

const Computer = ({setComputerModel}) => {
    const [count, setCount] = useState(1);
    const processors = [];
    for (let i = 0; i < count; i++) {
        processors.push(<Processor />);
    }
    
    return (
            <div className='group2'>
                <label className='form-label'>
                    Parts
                </label>
                {
                    processors
                }
                <div className='add-processor'><i className='processor-icon' fontSize="small" onClick={() => setCount(count + 1)}><AddOutlinedIcon /></i>Add processor</div>
            </div>
    )
}
export default Computer;