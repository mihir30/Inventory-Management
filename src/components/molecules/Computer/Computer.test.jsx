import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ReactDom from "react-dom";
import Computer from './Computer';

describe("Computer Component", () => {
    it('renders the component properly', () => {
        const divEl = document.createElement("div");
        ReactDom.render(<BrowserRouter><Computer setComputerModel={"PC-00019"}></Computer></BrowserRouter>, divEl); 
        
    });


    it('should have the right computer component snapshot', () => {
        const tree = renderer.create().toJSON();
        expect(tree).toMatchSnapshot();
    })

});