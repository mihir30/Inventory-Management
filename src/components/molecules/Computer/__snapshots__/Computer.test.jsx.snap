// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Computer Component should have the right computer component snapshot 1`] = `null`;
